waku/waku_archive/archive.nim

```diff
@@ -133,6 +133,9 @@ proc findMessages*(
   if query.contentTopics.len > 10:
     return err(ArchiveError.invalidQuery("too many content topics"))
 
+  if query.cursor.isSome() and query.cursor.get().hash.len != 32:
+    return err(ArchiveError.invalidQuery("invalid cursor"))
```

Comment:
Instead of doing this validation here, consider adding in `rcp_codec.nim`, line 103:
```nim
    if cursor.len != 32:
      return err(ProtobufError.missingRequiredField("cursor"))
```

Also, it's probably a good idea to replace line 94 for this, to validate the hashes length
```nim
   for i, buf in buffer:
      if buf.len != 32:
        return err(ProtobufError.invalidLengthField("message_hashes[" & $i & "]"))
```

Also, what do you think of editing `protocol.nim` to return the error detail when decoding a protobuffer:
```nim
    res.statusDesc = "decode rpc failed " & $error
```
For example with these changes the status description would look like this:
```
decode rpc failed (kind: MissingRequiredField, field: "cursor")
```

Reply:
> Instead of doing this validation here, consider adding in `rcp_codec.nim`, line 103:
> 
> ```nim
>     if cursor.len != 32:
>       return err(ProtobufError.invalidLengthField("cursor"))
> ```
> 
> Also, it's probably a good idea to replace line 94 for this, to validate the hashes length
> 
> ```nim
>    for i, buf in buffer:
>       if buf.len != 32:
>         return err(ProtobufError.invalidLengthField("message_hashes[" & $i & "]"))
> ```

I disagree, I feel like it would be best to keep cursor validation in archive and protobuf validation separate.

Is there a way to specify a limited `bytes` field in protobuf?

> What do you think of editing `protocol.nim` to return the error detail when decoding a protobuffer:
> 
> ```nim
>     res.statusDesc = "decode rpc failed " & $error
> ```
> 
> For example with these changes the status description would look like this:
> 
> ```
> decode rpc failed (kind: MissingRequiredField, field: "cursor")
> ```

:100 agree

Reply:
> Is there a way to specify a limited bytes field in protobuf?

I think it's not possible unless you use a code generator like https://github.com/bufbuild/protoc-gen-validate/tree/main which reads annotations from protobuffers. We're stuck with manual validation for now :disappointed: